x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/categorization_partition_field/categorization_stop_on_warn_switch.tsx

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useCallback, useContext, useEffect, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiSwitch } from '@elastic/eui';
+import { JobCreatorContext } from '../../../job_creator_context';
+import { AdvancedJobCreator, CategorizationJobCreator } from '../../../../../common/job_creator';
+
+export const CategorizationPerPartitionStopOnWarnSwitch: FC = () => {
+  const { jobCreator: jc, jobCreatorUpdate } = useContext(JobCreatorContext);
+  const jobCreator = jc as AdvancedJobCreator | CategorizationJobCreator;
+  const [stopOnWarn, setStopOnWarn] = useState(jobCreator.perPartitionStopOnWarn);
+
+  const toggleStopOnWarn = useCallback(() => setStopOnWarn(!stopOnWarn), [stopOnWarn]);
+
+  useEffect(() => {
+    jobCreator.perPartitionStopOnWarn = stopOnWarn;
+    jobCreatorUpdate();
+  }, [stopOnWarn]);
```

Comment:
this should also have a check in case the job json has been edited.
something like
```
useEffect(() => {
  setStopOnWarn (jobCreator.perPartitionStopOnWarn);
}, [jobCreatorUpdated]);
```

Reply:
Updated here https://github.com/elastic/kibana/commit/bf03137577f41383eff3299fba8a7e6d9dbca214